src/init.cpp

```diff
@@ -1847,8 +1832,21 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     }
 
 #if HAVE_SYSTEM
-    if (gArgs.IsArgSet("-blocknotify"))
+    if (args.IsArgSet("-blocknotify")) {
+        const std::string block_notify = args.GetArg("-blocknotify", "");
+        const auto BlockNotifyCallback = [block_notify](SynchronizationState sync_state, const CBlockIndex* pBlockIndex) {
+            if (sync_state != SynchronizationState::POST_INIT || !pBlockIndex)
+                return;
+
+            std::string strCmd = block_notify;
+            if (!strCmd.empty()) {
```

Comment:
fa33bc2dabbbd2d73961f9b0ce51420a3b6e4ad5

Why not check this before connecting `BlockNotifyCallback`?

Reply:
To keep the code move-only, as this commit is to be reviewed with `--color-moved=dimmed-zebra --color-moved-ws=ignore-all-space`

Reply:
Fair enough. I just think it would be fine to ignore that and make the change, even if in a separate commit. Otherwise I'd be happy to PR it if that's relevant.

Reply:
Feel free to ping me on your PR, once you have one